### PR TITLE
PYTHON-4530 - Move synchronized test code into top-level test directory

### PR DESCRIPTION
**Pull request by NoahStapp**
Once all the tests are converted, we will move them in a single commit to `test/synchronous`.

### CODE REVIEW

File: test/shared.py [hunk 1]
```diff
@@ -0,0 +1,367 @@
+# Copyright 2010-present MongoDB, Inc.
```
Comment (ShaneHarvey) [inline test/shared.py:1]: 2024.

Any alternative names for this file? “Shared.py” is pretty generic.

**Review by ShaneHarvey — COMMENTED** [review]
Nice!

File: test/shared.py (hunk 1 above)
Comment (NoahStapp) [reply test/shared.py:1]: What if we combine `test/utils.py` and this new file? Having multiple files that have effectively the same purpose seems unnecessary.

Comment (NoahStapp) [reply test/shared.py]: Renamed to `test/helpers.py` for now, we can revisit down the road.

File: test/__init__.py [hunk 2]
```diff
@@ -12,9 +12,10 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-"""Test suite for pymongo, bson, and gridfs."""
+"""Asynchronous test suite for pymongo, bson, and gridfs."""
```
Comment (blink1073) [inline test/__init__.py:15]: Isn't this synchronous?

**Review by blink1073 — APPROVED** [review]
LGTM!

**Pull request merged**